store/sqlstore/thread_store.go

```diff
@@ -702,22 +702,23 @@ func (s *SqlThreadStore) MaintainMembership(userId, postId string, opts store.Th
 			}
 			membership.LastUpdated = now
 			if _, err = s.updateMembership(trx, membership); err != nil {
-				return nil, err
+				return nil, false, err
 			}
 		}
 
 		if err = trx.Commit(); err != nil {
-			return nil, errors.Wrap(err, "commit_transaction")
+			return nil, false, errors.Wrap(err, "commit_transaction")
 		}
 
-		return membership, err
+		return membership, newMembership, err
```

Comment:
The reason this works is because of named return parameters. `newMembership` is defined in the return parameters. 
I feel this is confusing, we should just explicitly return `true` or `false`, same as in all the other returns in the function.